Packages/com.unity.inputsystem/InputSystem/Plugins/UI/InputSystemUIInputModule.cs

```diff
@@ -1897,6 +1897,13 @@ private void RemovePointerAtIndex(int index)
         {
             Debug.Assert(m_PointerStates[index].eventData.pointerEnter == null, "Pointer should have exited all objects before being removed");
 
+            // We don't want to release touch pointers on the same frame they are released. They get cleaned up one frame later in Process() 
```

Comment:
The reasoning in the comment makes sense to me, however I wonder why the below conditions are needed, it sounds like it would be more of a general fix that pointers (that may only exist during contact) shouldn't be removed mid frame for any scenario? 

I looked through the file but failed to spot where the clean-up/removal happens when this condition triggers, do you have a reference?

Reply:
Yes, the cleanup I'm referring to happens on line 2222 in the same file (InputSystemUIInputModule) in the Process() function call.

Reply:
Thanks